### PR TITLE
add documentation on search/replace

### PR DESCRIPTION
**Pull request by chrismiceli**
move notes to a link to reduce repeating information
replace some tabs with spaces
help content:
image
search replace dialog:
image

### CODE REVIEW

File: faq/prooffacehelp.php [hunk 1]
```diff
@@ -679,16 +679,42 @@ function echo_row( $name, $tooltip, $button_image_base, $accelerator )
 </p>
 
 
+<h3>Search/Replace Popup</h3>
```
Comment (cpeel) [inline faq/prooffacehelp.php]: We should consider adding an id to this `<h3>` and using it as an anchor in the help link.

File: tools/proofers/srchrep.php [hunk 2]
```diff
@@ -132,22 +99,7 @@ function preg_quote(str)
...
-// TRANSLATORS: Description of a{4} in a regular expression for an example
-echo "a{4} &mdash; " . _("four lowercase As") . "<br>\n";
-// TRANSLATORS: Description of [Aa]{6} in a regular expression for an example
-echo "[Aa]{6} &mdash; " . _("six As of either case") . "<br>\n";
-// TRANSLATORS: Description of A{2,8} in a regular expression for an example
-echo "A{2,8} &mdash; " . _("between 2 and 8 capital As") . "<br>\n";
-// TRANSLATORS: Description of [hb]e in a regular expression for an example
-echo "[hb]e &mdash; " . _("'he' or 'be'") . "<br>\n";
-?></p>
+<a href='<?php echo "$code_url/faq/prooffacehelp.php"; ?>' target='helpNewWin'><?php echo _('Regular Expression?'); ?></a>
```
Comment (cpeel) [inline tools/proofers/srchrep.php]: I realize you reused the prior text, but the link might be clearer if it was labeled 'Regular Expression Help'. @srjfoo, what do you think?

File: faq/prooffacehelp.php [hunk 3]
```diff
@@ -679,16 +679,42 @@ function echo_row( $name, $tooltip, $button_image_base, $accelerator )
...
+
+<p>
+    To make it easier to make repetative changes, this tool has been provided.
```
Comment (srjfoo) [inline faq/prooffacehelp.php]: repetitive

File: faq/prooffacehelp.php [hunk 4]
```diff
@@ -679,16 +679,42 @@ function echo_row( $name, $tooltip, $button_image_base, $accelerator )
...
+    Click on the Search/Replace button and a small window pops up,
+    containing a search value and a replace value.
+</p>
+<p>
+    Clicking <input type='button' value='Replace all.'> will replace all
+    all matched instances of the search text with the value from the replace text.
+    You may undo this change by clicking <input type='button' value='Undo.'>, but only
+    the most recent replace operation may be reverted.
```
Comment (srjfoo) [inline faq/prooffacehelp.php]: I realize that the instructions here are just matching the buttons, but, can we lose the periods terminating the buttons?

File: faq/prooffacehelp.php [hunk 5]
```diff
@@ -679,16 +679,42 @@ function echo_row( $name, $tooltip, $button_image_base, $accelerator )
...
+</p>
+<p>
+    For more complex search matching, regular expressions may be used.
+    <blockquote>
+        &mdash; any character<br>
```
Comment (srjfoo) [inline faq/prooffacehelp.php]: Shouldn't there be a dot representing "any character"? (There is in the dropdown that's part of the current s/r popup.)

File: tools/proofers/srchrep.php [hunk 6]
```diff
@@ -57,35 +57,6 @@ function test_and_enable_nonregex()
     }
 }
 
-function show_regex_help()
```
Comment (srjfoo) [inline tools/proofers/srchrep.php]: If this is getting removed from the popup, would it be a good idea to add a link to the s/r section of `prooffacehelp`?

File: tools/proofers/srchrep.php (hunk 2 above)
Comment (srjfoo) [reply tools/proofers/srchrep.php]: I agree.

File: faq/prooffacehelp.php [hunk 7]
```diff
@@ -679,16 +679,42 @@ function echo_row( $name, $tooltip, $button_image_base, $accelerator )
...
+<h3>Search/Replace Popup</h3>
+
+<p>
+    To make it easier to make repetative changes, this tool has been provided.
+    Click on the Search/Replace button and a small window pops up,
+    containing a search value and a replace value.
+</p>
+<p>
+    Clicking <input type='button' value='Replace all.'> will replace all
+    all matched instances of the search text with the value from the replace text.
```
Comment (srjfoo) [inline faq/prooffacehelp.php]: This displays as "replace all all matched"

**Comment by chrismiceli** [comment]
> 
> 
> I approve of all of the whitespace changes. Next time, please add the whitespace changes in a separate commit to break apart functional changes from code layout changes. Not worth modifying this MR for, however.

Yeah, sorry about that. I meant to commit it separately, then squashed it all locally before pushing.

File: tools/proofers/srchrep.php (hunk 6 above)
Comment (chrismiceli) [reply tools/proofers/srchrep.php]: You mean like hide the regular expression help and hide behind a javascript link?

**Comment by chrismiceli** [comment]
Updated screenshots:
image

image

File: tools/proofers/srchrep.php (hunk 6 above)
Comment (srjfoo) [reply tools/proofers/srchrep.php]: If that means move all the help to `prooffacehelp` and just link to the help there, yes. :) -- which it looks like you've done already

**Pull request merged**